src/lifecycles/unmount.js

```diff
@@ -3,50 +3,59 @@ import { handleAppError, transformErr } from '../applications/app-errors.js';
 import { reasonableTime } from '../applications/timeouts.js';
 import { getProps } from './prop.helpers.js';
 
-export async function toUnmountPromise(appOrParcel, hardFail = false) {
-  if (appOrParcel.status !== MOUNTED) {
-    return appOrParcel;
-  }
-  appOrParcel.status = UNMOUNTING;
+export function toUnmountPromise(appOrParcel, hardFail = false) {
```

Comment:
this one was the hardest one to get rid of async await in. the `finally {}` was sort of tricky. I would appreciate a detailed review of my work here, even though the tests indicate that it seems to (at least mostly) be working.